### PR TITLE
feat(postman-to-openapi): new postman-to-openapi package

### PR DESCRIPTION
**Pull request by Mason-Little**
Problem
Currently, there is no Postman spec to OpenAPI converter for our references

Solution
With this PR, I created a single-file TypeScript converter package in API references that allows us to convert any Postman spec JSON to OpenAPI JSON

### CODE REVIEW

**Comment by hanspagel** [comment]
This is awesome! I just moved the files around a little bit and added a README. I’ve also added one test, which fails for the Postman Collection example hosted on GitHub: https://github.com/postmanlabs/postman-collection/blob/develop/examples/collection-v2.json

Would you mind to look into this?

File: packages/postman-to-openapi/rollup.config.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { createRollupConfig } from '@scalar/build-tooling'
+import type { RollupOptions } from 'rollup'
+
+const options: RollupOptions = {
+  input: ['./src/index.ts'],
+  ...createRollupConfig({
+    // Needed to enable explicit TypeScript (which Vite does not need)
+    typescript: true,
+  }),
+}
```
Comment (geoffgscott) [inline packages/postman-to-openapi/rollup.config.ts:10]: No destructuring needed
```suggestion
const config = createRollupConfig({
  options: {
    input: ['src/index.ts'],
  },
  typescript: true,
})
```

File: packages/postman-to-openapi/src/fixtures/collection-v2.json
Comment (geoffgscott) [inline packages/postman-to-openapi/src/fixtures/collection-v2.json:1]: Test fixtures should go in a `test` folder adjacent to the `src` folder. This allows easy TSConfig exclusion of all test files.

File: packages/postman-to-openapi/env.d.ts
Comment (geoffgscott) [inline packages/postman-to-openapi/env.d.ts:1]: We should not be using raw imports unless we using Vite in which case the vite shims will manage this.

File: packages/postman-to-openapi/src/convert.ts [hunk 2]
```diff
@@ -0,0 +1,771 @@
...
+
+function replacePostmanVariables(collectionFile: any): any {
+  return collectionFile
+}
+/**
+ * Converts a Postman collection to an OpenAPI specification.
+ *
+ * @param {string | Record<string, any>} collectionContent - The Postman collection content, either as a JSON string or an object.
+ * @param {Options} [options={}] - Configuration options for the conversion process.
+ * @returns {Promise<string>} A promise that resolves to the OpenAPI specification as a JSON string.
```
Comment (geoffgscott) [inline packages/postman-to-openapi/src/convert.ts:24]: We don't need JSDocs for anything as we can directly annotate parameters using typescript.

File: packages/postman-to-openapi/src/convert.ts [hunk 3]
```diff
@@ -0,0 +1,771 @@
...
+  if (typeof collectionContent === 'string') {
+    collectionFile = JSON.parse(collectionContent)
+  } else {
+    collectionFile = collectionContent
+  }
+  if (replaceVars) {
+    collectionFile = replacePostmanVariables(collectionFile)
+  }
+  const postmanJson = collectionFile.collection || collectionFile
+  const { item: items = [], variable = [] } = postmanJson
```
Comment (geoffgscott) [inline packages/postman-to-openapi/src/convert.ts:54]: Please do a generator function pattern here and not `let` reassignment with added destructuring

File: packages/postman-to-openapi/src/convert.ts [hunk 4]
```diff
@@ -0,0 +1,771 @@
...
+  const paths: Record<string, any> = {}
+  const domains = new Set<string>()
+  const tags: Record<string, any> = {}
+  const securitySchemes: Record<string, any> = {}
+
+  for (let i = 0; i < items.length; i++) {
+    let element = items[i]
```
Comment (geoffgscott) [inline packages/postman-to-openapi/src/convert.ts:61]: Prefer `items.forEach`

File: packages/postman-to-openapi/src/convert.ts [hunk 5]
```diff
@@ -0,0 +1,771 @@
...
+    while (element?.item != null) {
+      const { item = [], description: tagDesc = '' } = element
+      const tag = calculateFolderTag(element, folders)
+      const tagged = item.map((e: any) => ({ ...e, tag }))
+      tags[tag] = tagDesc
+      items.splice(i, 1, ...tagged)
+      element = tagged.length > 0 ? tagged.shift() : items[i]
+    }
```
Comment (geoffgscott) [inline packages/postman-to-openapi/src/convert.ts:69]: While loops can be difficult to follow and can have undefined exit conditions. 
We are also mutating the array we are iterating over here (items) which will cause unintended side effects (and is a bad code smell).

File: packages/postman-to-openapi/src/convert.ts [hunk 6]
```diff
@@ -0,0 +1,771 @@
...
+        response,
+      } = element
+      const {
+        url,
+        method,
+        body,
+        description: rawDesc,
+        header = [],
+        auth,
+      } = request
```
Comment (geoffgscott) [inline packages/postman-to-openapi/src/convert.ts:85]: This feels like excessive destructuring. nothing wrong with using `element.name` which in a large function block like this will be more clear anyways.

File: packages/postman-to-openapi/src/convert.ts [hunk 7]
```diff
@@ -0,0 +1,771 @@
...
+    ...parseExternalDocs(variable, externalDocs),
+    ...parseServers(domains, servers),
+    ...parseAuth(postmanJson, optsAuth, securitySchemes),
+    ...parseTags(tags),
+    paths,
+  }
+
+  return JSON.stringify(openApi, null, 4)
+}
+/* Helper Functions */
```
Comment (geoffgscott) [inline packages/postman-to-openapi/src/convert.ts:129]: Incorrect comment syntax for a function. If you want a module brake using the VSCode snippet `//`

Comment (geoffgscott) [reply packages/postman-to-openapi/src/convert.ts:129]: This module file is also getting quite long. Would suggest creating some separate module files to group some of these logically. All of the below functions need unit tests.

File: packages/postman-to-openapi/src/convert.ts [hunk 8]
```diff
@@ -0,0 +1,771 @@
...
+    ? { externalDocs: { url, ...(description ? { description } : {}) } }
+    : {}
+}
+
+function parseBody(body: any = {}, method: string): any {
+  if (['GET', 'DELETE'].includes(method)) return {}
+  const { mode, raw, options = { raw } } = body
+  let content: any = {}
+  switch (mode) {
+    case 'raw': {
```
Comment (geoffgscott) [inline packages/postman-to-openapi/src/convert.ts:223]: No `switch` and no `let`. In the scalar codestyles you can see examples that use the early return pattern which is much easier to follow. 

```ts
if(conditionOne) return dataFormOne
if(conditionTwo) return dataFormTwo
```

File: packages/postman-to-openapi/src/convert.ts [hunk 9]
```diff
@@ -0,0 +1,771 @@
...
+function parseFormData(data: any[]): any {
+  const objectSchema: any = {
+    schema: {
+      type: 'object',
+    },
+  }
+  return data.reduce((obj: any, { key, type, description, value }: any) => {
+    const { schema } = obj
+    if (isRequired(description)) {
+      ;(schema.required = schema.required || []).push(key)
```
Comment (geoffgscott) [inline packages/postman-to-openapi/src/convert.ts:293]: This is hard to parse syntax. This can be broken into two lines or done differently.

File: packages/postman-to-openapi/src/convert.ts [hunk 10]
```diff
@@ -0,0 +1,771 @@
...
+    }
+    ;(schema.properties = schema.properties || {})[key] = {
```
Comment (geoffgscott) [inline packages/postman-to-openapi/src/convert.ts:295]: Same as above comment

File: packages/postman-to-openapi/src/convert.ts [hunk 11]
```diff
@@ -0,0 +1,771 @@
...
+): any {
+  const parameters = [
+    ...Array.from(
+      Array.isArray(header) ? 
+        header.reduce(
+          mapParameters('header', includeHeader, paramInserter),
+          new Map(),
+        ) : [] // If not an array, default to an empty array
+    ),
+  ]
```
Comment (geoffgscott) [inline packages/postman-to-openapi/src/convert.ts:327]: This needs some clean up. To hard to follow.

File: packages/postman-to-openapi/src/convert.ts [hunk 12]
```diff
@@ -0,0 +1,771 @@
...
+  header: any[] = [],
+  paths: string,
+  paramsMeta: any = {},
+  pathVars: any = {},
+  { includeQuery = false, includeHeader = false }: any = {},
+  paramInserter: (
+    parameterMap: Map<string, any>,
+    param: any,
+  ) => Map<string, any> = defaultParamInserter,
+): any {
```
Comment (geoffgscott) [inline packages/postman-to-openapi/src/convert.ts:318]: No ts `any`

File: packages/postman-to-openapi/src/convert.ts [hunk 13]
```diff
@@ -0,0 +1,771 @@
...
+}
+
+function mapParameters(
+  type: string,
+  includeDisabled: boolean,
+  paramInserter: (
+    parameterMap: Map<string, any>,
+    param: any,
+  ) => Map<string, any>,
+): (parameterMap: Map<string, any>, param: any) => Map<string, any> {
```
Comment (geoffgscott) [inline packages/postman-to-openapi/src/convert.ts:345]: No ts `any` unless there is a very strong reason and it's well commented.

**Comment by Mason-Little** [comment]
Marc Amrit and I decided it would be better to restart after syncing on this PR
